Subject: [PATCH] Use GNUInstallDirs.

x32 ABI build fails because of improper library path. Use
GNUInstallDirs to fix it.

Upstream-Status: Backport

Signed-off-by: Maxin B. John <maxin.john@intel.com>
---
diff -Naur libical-1.0.1-orig/CMakeLists.txt libical-1.0.1/CMakeLists.txt
--- libical-1.0.1-orig/CMakeLists.txt	2015-12-07 18:13:58.311503974 +0200
+++ libical-1.0.1/CMakeLists.txt	2015-12-07 18:17:18.362002998 +0200
@@ -92,6 +92,10 @@
 endif()
 
 set(PERL_EXECUTABLE perl)
+# Ensure finding 64bit libs when using 64-bit compilers
+if(CMAKE_CL_64)
+  set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIB64_PATHS True)
+endif()
 
 # MSVC specific definitions
 if(WIN32)
@@ -104,10 +108,10 @@
   add_definitions(-DBIG_ENDIAN=0 -DLITTLE_ENDIAN=1 -DBYTE_ORDER=BIG_ENDIAN)
 endif()
 
-set(LIB_SUFFIX "" CACHE STRING "Define suffix of directory name (32/64)")
-set(LIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}" CACHE STRING "Library directory name" FORCE)
-set(INCLUDE_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/include" CACHE STRING "Include directory name" FORCE)
-set(SHARE_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/share" CACHE STRING "Share directory name" FORCE)
+include(GNUInstallDirs)
+set(LIB_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR} CACHE STRING "Library directory name" FORCE)
+set(INCLUDE_INSTALL_DIR ${CMAKE_INSTALL_INCLUDEDIR} CACHE STRING "Include directory name" FORCE)
+set(SHARE_INSTALL_DIR ${CMAKE_INSTALL_DATAROOTDIR} CACHE STRING "Share directory name")
 
 # set the output paths
 set(EXECUTABLE_OUTPUT_PATH ${CMAKE_BINARY_DIR}/bin)
